app/src/main/java/com/example/countrylist/countriesList/activity/MainActivity.kt

```diff
@@ -0,0 +1,82 @@
+package com.example.countrylist.countriesList.activity
+
+import android.content.Intent
+import android.os.Bundle
+import android.widget.ProgressBar
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.example.countrylist.CountriesListQuery
+import com.example.countrylist.R
+import com.example.countrylist.base.constants.Utils
+import com.example.countrylist.base.repository.implementation.CountryRepositoryImpl
+import com.example.countrylist.countriesList.adapter.CountryAdapter
+import com.example.countrylist.countriesList.contract.MainActivityContract
+import com.example.countrylist.countriesList.presenter.MainActivityPresenter
+import com.example.countrylist.countryDetails.activity.DetailsActivity
+
+class MainActivity : AppCompatActivity(), CountryAdapter.RVOnClickListener,
+    MainActivityContract.View {
+
+    private var countriesList: MutableList<CountriesListQuery.Country> = ArrayList()
```

Comment:
View should not be aware about countriesList - it is presenter responsibility.
Can we make presenter implement CountryAdapter.RVOnClickListener interface?